pkg/reconciler/apis/apibinding/apibinding_reconcile.go

```diff
@@ -483,32 +453,43 @@ func crdFromAPIResourceSchema(schema *apisv1alpha1.APIResourceSchema) (*apiexten
 	return crd, nil
 }
 
-// lockedStringSet guards a sets.String with an RWMutex.
-type lockedStringSet struct {
-	lock sync.RWMutex
-	s    sets.String
+func getAPIExportClusterName(apiBinding *apisv1alpha1.APIBinding) (logicalcluster.LogicalCluster, error) {
+	parent, hasParent := logicalcluster.From(apiBinding).Parent()
+	if !hasParent {
+		return logicalcluster.LogicalCluster{}, fmt.Errorf("a APIBinding in %s cannot reference a workspace", logicalcluster.From(apiBinding))
+	}
+	if apiBinding.Spec.Reference.Workspace == nil {
+		return logicalcluster.LogicalCluster{}, fmt.Errorf("APIBinding does not specify an APIExport")
```

Comment:
`// cannot happen due to APIBinding validation`

Reply:
Fixed